src/netaddress.cpp

```diff
@@ -271,6 +275,30 @@ bool CNetAddr::SetSpecial(const std::string& str)
     return false;
 }
 
+bool CNetAddr::SetI2P(const std::string& str)
+{
+    static const char* suffix{".b32.i2p"};
+    static constexpr size_t suffix_len{8};
+
+    if (!ValidAsCString(str) || str.size() <= suffix_len ||
+        str.substr(str.size() - suffix_len) != suffix) {
+        return false;
+    }
+
```

Comment:
There are (or there will be) more than one base32 encoded addresses, or at least that's what I found reading the i2p specs. The "traditional" one is 52 chars length while the `Encrypted LS with Base 32 Addresses` is 56 chars length (https://geti2p.net/spec/encryptedleaseset.txt)

It seems we only support the 32 bytes addresses (52 chars) so, it could make sense to validate it.
 
```suggestion
    // base32-encoded public key hash (SHA256) length (without padding) 
    if ((str.size() - suffix_len) != 52)
    {
        return false;
    }
```

Reply:
We already do that below: `address_bytes.size() != ADDR_I2P_SIZE`. It would be a minuscule optimization to invalidate it before Base32-decoding, but I will skip it for now (same applies when parsing Tor addresses).

Reply:
Make sense.

Reply:
Added such a check, see https://github.com/bitcoin/bitcoin/pull/20685#discussion_r549161698 for details.